### PR TITLE
motr/client: fix some index ops api documentation

### PR DESCRIPTION
**Pull request by andriytk**
Fix wrong and misleading descriptions of some index operations, especially M0_IC_LIST.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by siningwuseagate — APPROVED** [review]

**Review by nikitadanilov — APPROVED** [review]

File: motr/client.h [hunk 1]
```diff
@@ -554,11 +554,11 @@ enum m0_idx_opcode {
...
 	/** Delete the value, if any, for the given key. */
 	M0_IC_DEL,                 /* 17 */
-	/** Given a key, return the next key and its value. */
+	/** Given a key, return the next keys and their values. */
 	M0_IC_NEXT,                /* 18 */
-	/** Check an index for an existence. */
+	/** Check index for the given key existence. */
 	M0_IC_LOOKUP,              /* 19 */
-	/** Given an index id, get the list of next indices. */
+	/** Given a key, return the list of next keys. */
```
Comment (huanghua78) [inline motr/client.h:561]: I don't think this is correct.

This operation is to enumerate indices, not the records of an index.
It is like "getdir()", from this index, to list a predefined number of indices next to this one.

**Review by huanghua78 — CHANGES_REQUESTED** [review]

File: motr/client.h (hunk 1 above)
Comment (andriytk) [reply motr/client.h:561]: Fixed.

Comment (huanghua78) [reply motr/client.h:561]: I still think the original document is better.

Comment (andriytk) [reply motr/client.h:561]: For which op?

Comment (huanghua78) [reply motr/client.h:561]: M0_IC_LIST

Comment (andriytk) [reply motr/client.h:561]: Sorry, but it was just wrong! `M0_IC_LIST` returns the list of keys from an index (given the starting point key), not the list of indexes.

Comment (huanghua78) [reply motr/client.h:561]: No, this is wrong.

You can refer to this: https://github.com/Seagate/cortx-motr/pull/1651
It has some example output of "index list".

Comment (andriytk) [reply motr/client.h:561]: I see...

Unfortunately, this PR was already merged by accident, so here is the fix - https://github.com/Seagate/cortx-motr/pull/1653.

**Pull request merged**